display_list/dl_builder.cc

```diff
@@ -108,12 +102,13 @@ sk_sp<DisplayList> DisplayListBuilder::Build() {
   save_stack_.pop_back();
   Init(rtree != nullptr);
 
-  storage_.realloc(bytes);
+  storage_.trim();
+  offsets_.shrink_to_fit();
```

Comment:
I probably would not use shrink_to_fit here since it may imply another O(N) copy.

Reply:
Is there a shrink on copy then?

Reply:
We've always been trimming the storage, though.

How often is realloc(smaller) another allocation and copy?

Reply:
I don't know about realloc, I just mean for the std::vector.

from https://cplusplus.com/reference/vector/vector/shrink_to_fit/

> This may cause a reallocation, but has no effect on the [vector size](https://cplusplus.com/vector::size) and cannot alter its elements.

Reply:
With the change to swap mentioned in the other comment, I still trim the storage because we used to do that, but not the offsets since we have no back history of that.

I can eliminate the storage trim in a separate PR so we can track that independently.

I can also investigate if trimming the offsets makes a difference. `std::vector` seems to use a doubling algorithm which means we could be using twice the storage for the vectors. I'm still not convinced that shrinking it will require an allocation...

Reply:
shrinking it may require an allocation, as the reference says. 🤷‍♂️